### PR TITLE
Freshsales suite sample app with Entity Storage

### PR DESCRIPTION
**Pull request by zachjonesnoel**
- Added sample app for Freshsales suite.
- Updated the UI for contact entity page app.
- Added screenshots for different scenarios
- Update ReadMe

### CODE REVIEW

File: freshsales-suite/contact-kyc/config/iparams.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "fcrm_domain": {
+        "display_name": "Freshsales Domain",
+        "description": "Please enter your Freshsales domain web address",
+        "type": "text",
```
Comment (kaustavdm) [inline freshsales-suite/contact-kyc/config/iparams.json]: Use either `"type": "domain"` or `"type": "url"`.

File: freshsales-suite/contact-kyc/app/scripts/contact_entity_menu.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    headers: {
+      Authorization: `Token token=${iparams.fcrm_api_key}`
+    }
+  }
+  return await client.request.get(url, options);
+}
+
+async function createContactEntity(client, contactDetails) {
+  let contact_entity = await contacts(client);
+  contact_entity.create({
```
Comment (kaustavdm) [inline freshsales-suite/contact-kyc/app/scripts/contact_entity_menu.js]: `await` this. Or return the `Promise` to the caller.

File: freshsales-suite/contact-kyc/app/scripts/contact_entity_menu.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+};
+
+async function getContact(client, currentEntityId, iparams) {
+  const url = `${iparams.fcrm_domain}api/contacts/${currentEntityId}`;
+  const options = {
+    headers: {
+      Authorization: `Token token=${iparams.fcrm_api_key}`
+    }
+  }
+  return await client.request.get(url, options);
```
Comment (kaustavdm) [inline freshsales-suite/contact-kyc/app/scripts/contact_entity_menu.js]: Don't need `await` here. The caller can `await` the returned Promise.

File: freshsales-suite/contact-kyc/app/scripts/contact_entity_menu.js [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+        client.events.on("app.activated", function () {
+          onAppActivate(client);
+        });
+      })
+      .catch(handleErr);
+  }
+};
+
+async function getContact(client, currentEntityId, iparams) {
+  const url = `${iparams.fcrm_domain}api/contacts/${currentEntityId}`;
```
Comment (kaustavdm) [inline freshsales-suite/contact-kyc/app/scripts/contact_entity_menu.js]: Change this based on the `type` chosen for the `fcrm_domain` field in iparams.

File: freshsales-suite/contact-kyc/manifest.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        },
+        "full_page_app": {
+          "url": "kyc_pending_fullpage_app.html",
+          "icon": "images/icon.svg"
+        }
+      }
+    }
+  },
+  "whitelisted-domains": [
+    "https://*.freshworks.com"
```
Comment (kaustavdm) [inline freshsales-suite/contact-kyc/manifest.json]: This should probably be `"*.myfreshworks.com"`. Or, you can use template substitution for the `fcrm_domain` iparam like this: `<%= iparams.fcrm_domain %>`

**Review by kaustavdm — CHANGES_REQUESTED** [review]
Few small changes needed.

File: freshsales-suite/contact-kyc/config/iparams.json (hunk 1 above)
Comment (zachjonesnoel) [reply freshsales-suite/contact-kyc/config/iparams.json]: Have used `"type":"url"` as this is an omni app would have to handle for both *Freshsales* and *Freshworks_crm*.

File: freshsales-suite/contact-kyc/app/scripts/contact_entity_menu.js (hunk 2 above)
Comment (zachjonesnoel) [reply freshsales-suite/contact-kyc/app/scripts/contact_entity_menu.js]: awaited the entity create.

File: freshsales-suite/contact-kyc/app/scripts/contact_entity_menu.js (hunk 3 above)
Comment (zachjonesnoel) [reply freshsales-suite/contact-kyc/app/scripts/contact_entity_menu.js]: Changed and also added try-catch.

File: freshsales-suite/contact-kyc/app/scripts/contact_entity_menu.js (hunk 4 above)
Comment (zachjonesnoel) [reply freshsales-suite/contact-kyc/app/scripts/contact_entity_menu.js]: `url` returns as the string value itself so retained this.

File: freshsales-suite/contact-kyc/manifest.json (hunk 5 above)
Comment (zachjonesnoel) [reply freshsales-suite/contact-kyc/manifest.json]: Changed to the URL from iparam - `<%= iparam.fcrm_domain %>`

File: freshsales-suite/contact-kyc/config/iparams.json (hunk 1 above)
Comment (zachjonesnoel) [reply freshsales-suite/contact-kyc/config/iparams.json]: The `fcrm_api_key` is retained as text as again if we change the type to `api_key`, it needs to handle an associated product and in this case it's an omniapp.

**Review by kaustavdm — APPROVED** [review]
LGTM overall. Thanks for doing this @zachjonesnoel!

**Pull request merged**